### PR TITLE
Updates

### LINKED ISSUE #234: Parsing error
### Did you check existing issues?

- [X] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [X] I have searched the existing issues of tree-sitter-php

### Tree-Sitter CLI Version, if relevant (output of `tree-sitter --version`)

tree-sitter 0.21.0

### Describe the bug

I saw the following valid PHP code in the wild that was parsed incorrectly. I was able to extract the part that wasn't parsed properly into a smaller valid PHP example that is reproducible with tree-sitter cli.

### Steps To Reproduce/Bad Parse Tree

```
isla@Islas-MacBook-Pro ~/source/tree-sitter-php/php (master *%|u=) $ tree-sitter parse test.php
running: "c++" "--version"
exit status: 0
running: "xcrun" "--show-sdk-platform-version" "--sdk" "macosx"
exit status: 0
(program [0, 0] - [10, 0]
  (php_tag [0, 0] - [0, 5])
  (expression_statement [1, 0] - [1, 14]
    (assignment_expression [1, 0] - [1, 13]
      left: (variable_name [1, 0] - [1, 4]
        (name [1, 1] - [1, 4]))
      right: (encapsed_string [1, 7] - [1, 13]
        (string_value [1, 8] - [1, 12]))))
  (ERROR [2, 0] - [10, 0]
    (function_call_expression [2, 4] - [2, 50]
      function: (name [2, 4] - [2, 14])
      arguments: (arguments [2, 14] - [2, 50]
        (argument [2, 15] - [2, 43]
          (string [2, 15] - [2, 43]
            (string_value [2, 16] - [2, 42])))
        (argument [2, 45] - [2, 49]
          (variable_name [2, 45] - [2, 49]
            (name [2, 46] - [2, 49])))))
    (name [3, 7] - [3, 17])
    (comment [3, 19] - [3, 65])
    (string_value [3, 65] - [10, 0])))
```

### Expected Behavior/Parse Tree

Both of the conditionals should be parsed the same

### Repro

```PHP
<?php
$var = "test";
if (preg_match('#([\w\(\)\.\,\;]|[`]{1})$#', $var)
    && preg_match('#^([\w\(\)\.\,\;`]|\\\--[\w]|[`]{1})#', $var) 
) {
    echo "test";
}
?>
```

### PR DESCRIPTION
**Pull request by amaanq**
Closes #234

### CODE REVIEW

**Comment by calebdw** [comment]
@amaanq, thanks!

I guess I was missing the `token.immediate` in the `string_content` rule...


I have a few questions though:
- it looks like you removed the parse examples script...is there any way to run this outside of ci?
- there seems to be a lot of `string_contents` in the added testcase:
  ```scheme
              (string
                (string_content)
                (string_content)
                (string_content)
                (string_content)
                (string_content)
                (string_content)
                (string_content)
                (escape_sequence)
                (string_content)
                (string_content))
  ```
  but should they not be condensed into a single node on either side of the escape sequence?
  ```scheme
              (string
                (string_content)
                (escape_sequence)
                (string_content))
  ```
- it looks like all of the build scripts in the package.json were replaced by the `prebuildify`? If so that's pretty neat

**Comment by amaanq** [comment]
huh good point, I'll look at the string stuff in a bit

as to running examples outside of ci, no not really, since it was just cluttering the repo imo - if CI fails there's an option to make it upload the failed files (check example in parse-action repo), and so you can just download it from there - otherwise for local dev you can just copy the git clone commands in the ci file, and run ts p examples/**/*.php -q

yeah prebuildify is very nice, esp now that we've moved to napi from nan

File: common/scanner.h [hunk 1]
```diff
@@ -239,20 +197,27 @@ static inline bool scan_nowdoc_string(Scanner *scanner, TSLexer *lexer) {
...
         }
         advance(lexer);
         has_consumed_content = true;
 
-        end_tag_matched = (i == heredoc_tag.len - 1 && (iswspace(lexer->lookahead) || lexer->lookahead == ';' ||
-                                                        lexer->lookahead == ',' || lexer->lookahead == ')'));
+        array_push(&word, heredoc_tag.contents[i]);
+    }
+
+    if (valid_symbols[HEREDOC_END] && string_eq(&word, &heredoc_tag)) {
```
Comment (calebdw) [inline common/scanner.h:214]: @amaanq, this might need to be NOWDOC_END

**Pull request merged**